tests/test_geometry.py

```diff
@@ -361,3 +361,30 @@ def test_create_CustomKL_geometry():
 	   geom.trunc_term==trunc_term
 	
 
+def test_KLExpansion_projection():
+    """Check KLExpansion geometry projection performed by the method fun2par)"""
+    # Set up a KLExpansion geometry
+    num_modes = 95
+    N = 100
+    L = 1.0
+    grid = np.linspace(0,1,N)
+
+    geom = cuqi.geometry.KLExpansion(grid, num_modes=num_modes,
+     decay_rate=1.5,
+     normalizer=12.0)
+
+    # Create a signal 
+    signal =1/30*(1-np.cos(2*np.pi*(L-grid)/(L)))\
+                    +1/30*np.exp(-2*(10*(grid-0.5))**2)+\
+                     1/30*np.exp(-2*(10*(grid-0.8))**2)
+
+    # Project signal to the KL basis and back
+    p = geom.fun2par(signal)
+    assert(len(p) == num_modes)
+
+    signal_proj = geom.par2fun(p)
+    assert(len(signal_proj) == N)
+
+    # Check that the projection is accurate
+    rel_err = np.linalg.norm(signal-signal_proj)/np.linalg.norm(signal)
+    assert np.isclose(rel_err, 0.0, atol=1e-5)
```

Comment:
Just to check my understanding, this test uses a signal that can be perfectly represented in the KLExpansion, and as such there is no loss in projecting back and forth? Good to test that, and I think it would be relevant to add at least one test of a signal that can NOT be perfectly represented - one needs to work out how to check for correctness since cannot simply compare with orignal.

Reply:
Thank you for brining this up. Here it is not exact transformation. If the number of modes is 100 I believe it will be. The lower the number of modes the further the par2fun(fun2par(true_signal)) from the true_signal. Here I choose the number of modes to be 95, which is close to 100. and I adjusted the isclose tolerance to pass the assertion.